AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -168,11 +169,13 @@ - (NSArray *)children
 
 - (ASEnvironmentState)environmentState
 {
+  ASDN::MutexLocker l(_propertyLock);
```

Comment:
I am not certain this is safe.  I believe there is a deadlock risk here - let's reference the blame in this file for the change that had to remove a certain lock.  @rcancro @garrettmoon @maicki